lumicks/pylake/kymotracker/tests/test_kymotrack.py

```diff
@@ -116,15 +117,88 @@ def test_kymotrackgroup(blank_kymo):
     assert len(tracks) == 4
     assert tracks[0] == k1
     assert tracks[1] == k2
+    assert tracks[-1] == k4
+    assert len(tracks[0:2]) == 2
     assert tracks[0:2][0] == k1
     assert tracks[0:2][1] == k2
+    assert tracks[1:3][0] == k2
+    assert tracks[1:3][1] == k3
+    assert tracks[2:-1][0] == k3
+    assert len(tracks[2:-1]) == 1
+    assert tracks[2:][0] == k3
+    assert tracks[2:][1] == k4
 
     with pytest.raises(IndexError):
         tracks[0:2][2]
 
     with pytest.raises(NotImplementedError):
         tracks[1] = 4
 
+
+@pytest.mark.parametrize(
+    "remove,remaining",
+    [
+        ([1], [True, False, True]),
+        ([0, 1], [False, False, True]),
+        ([2, 1], [True, False, False]),
+    ],
+)
+def test_kymotrackgroup_remove(blank_kymo, remove, remaining):
+    src_tracks = [
+        KymoTrack(np.array([1, 2, 3]), np.array([2, 3, 4]), blank_kymo, "red"),
+        KymoTrack(np.array([2, 3, 4]), np.array([3, 4, 5]), blank_kymo, "red"),
+        KymoTrack(np.array([3, 4, 5]), np.array([4, 5, 6]), blank_kymo, "red"),
+    ]
+
+    tracks = KymoTrackGroup(src_tracks)
+    for track in remove:
+        tracks.remove(src_tracks[track])
+    for track, should_be_present in zip(src_tracks, remaining):
+        if remaining:
+            assert track in tracks
+        else:
+            assert track not in tracks
+
+
+def test_kymotrackgroup_lookup(blank_kymo):
+    k1 = KymoTrack(np.array([1, 2, 3]), np.array([2, 3, 4]), blank_kymo, "red")
+    k2 = KymoTrack(np.array([2, 3, 4]), np.array([3, 4, 5]), blank_kymo, "red")
+    k3 = KymoTrack(np.array([3, 4, 5]), np.array([4, 5, 6]), blank_kymo, "red")
+
+    tracks = KymoTrackGroup([k1, k2, k3])
+    tracks._get_track_by_id(id(k1)) is k1
+    tracks._get_track_by_id(id(k2)) is k2
```

Comment:
Can we test with a KymoTrackGroup that contains the same track multiple times, or a track that is shared between at least two KymoTrackGroups?

Reply:
The former should be explicitly excluded now.

Anything specific you would like to test with the track that is shared between two groups?